### ISSUE TITLE
Applying x-ms-header-collection-prefix on headers is not propagated to code generators

### ISSUE THREAD

**Issue by jhendrixMSFT** [Azure/autorest.modelerfour#337]
**Describe the issue**
Regarding https://github.com/Azure/azure-rest-api-specs/blob/storage-dataplane-preview/specification/storage/data-plane/Microsoft.BlobStorage/preview/2019-07-07/blob.json

Specifically this.

```json
            "headers": {
              "x-ms-meta": {
                "type": "string",
                "x-ms-client-name": "Metadata",
                "x-ms-header-collection-prefix": "x-ms-meta-"
              },
```

The `x-ms-header-collection-prefix` isn't exposed on the underlying header type in the code model.

** REQUIRED ** 
**ModelerFour** version: 4.15.407
**AutoRest-core** version: 3.0.6306
**CommandLine** :

autorest --use=\git\Azure\autorest.go --license-header=MICROSOFT_MIT_NO_VERSION --openapi-type=data-plane --file-prefix="zz_generated_" --export-client="false" --credential-scope="https://storage.azure.com/.default" --input-file=https://raw.githubusercontent.com/Azure/azure-rest-api-specs/storage-dataplane-preview/specification/storage/data-plane/Microsoft.BlobStorage/preview/2019-07-07/blob.json --module=azstorage --output-folder=\git\Azure\autorest.go\test\storage\2019-07-07\azblob

OpenAPI/Swagger file:
https://github.com/Azure/azure-rest-api-specs/blob/storage-dataplane-preview/specification/storage/data-plane/Microsoft.BlobStorage/preview/2019-07-07/blob.json

**Comment by daviwil** [comment]
The fix for this is now released in `@autorest/modelerfour@4.15.419`, please give it a try and let me know if it resolves the issue.

**Comment by jhendrixMSFT** [comment]
Fix verified, thanks!
